pandas/tests/io/test_parquet.py

```diff
@@ -724,6 +744,13 @@ def test_timestamp_nanoseconds(self, pa):
         df = pd.DataFrame({"a": pd.date_range("2017-01-01", freq="1n", periods=10)})
         check_round_trip(df, pa, write_kwargs={"version": "2.0"})
 
+    def test_timezone_aware_index(self, pa, timezone_aware_date_list):
```

Comment:
can you change this so that on future versions of pyarrow it will set check_dtype=True (IOW when the bug is fixed). otherwise I am worried this will perpetuate forever.

alternatively, could remove the check_dtype and just xfail this (and when new pyarrow fixes this test will then start to xpass which will make it fail as we have strict=True). so again could make this a conditional on an older / newer version of pyarrow.

Reply:
@jreback based on the comment above by @jorisvandenbossche , I don't expect this to be "fixed" in Arrow. `pytz.FixedOffset` is a wrapper around `timedelta` like `datetime.timezone` is and they both implement `datetime.tzinfo`. ofc I'm happy to do any change you ask

Reply:
Yes, there are no active plans in pyarrow to change this. Although it makes probably sense to use `datetime.timezone`, but if we do that, it's `pytz.FixedOffset` that won't be preserved, in which case another test probably needs a `check_dtype=False`.

Reply:
@jreback I added a long description. I know this won't help a follow up improvement in the future, but at this point it's unlikely to change.

There are two scenarios:
1. arrow/parquet persists the initial class information
2. the assert_ introduces a sub-feature of check_dtype=True (`check_timezone_dtype=False` by default) accepting `pytz.FixedOffset` and `datetime.timezone` equality

I don't see any of them on the short or mid-term roadmap.